enclave/core/sgx/hostcalls.c

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "internal_t.h"
+
+void* oe_host_realloc(void* ptr, size_t size)
+{
+    void* retval = NULL;
+
+    if (!ptr)
+        return oe_host_malloc(size);
+
+    if (oe_realloc_ocall(&retval, ptr, size) != OE_OK)
```

Comment:
We need ensure that oe_is_outside_enclave(retval, size) is true. Otherwise the host can handback a pointer to enclave memory and get the enclave to overwrite its contents.

Reply:
Good catch. Resolved.